### ISSUE TITLE
kubectl plugin - reference to current directory

### ISSUE THREAD

**Issue by ae-dg** [kubernetes/kubernetes#55708]
**Is this a BUG REPORT or FEATURE REQUEST?**: feature request

/kind feature

When executing a plugin with a "kubectl plugin" command the current directory is changed to the directory of the plugin, but there is no reference to the directory the user was in before.

It would be nice to have a KUBECTL_PLUGIN variables set to the directory the user was in to be able to access files in there.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.8.0
- Kubectl version: 1.8.2
- Cloud provider or hardware configuration: minikube
- OS (e.g. from /etc/os-release): macOS Sierra 10.12.6

**Comment by jordanwilson230** [comment]
+1 ...I've run into the same issue. It must be an issue like subshell inheritance. Only when running a command directly from the plugin.yaml file (i.e., echo $PWD) does it return the user's directory. Once the plugin.yml launches the script/binary, then the user's current environment is lost.

Without access to this, the plugin feature becomes unsuitable for many (if not all) of our use cases.

**Comment by ae-dg** [comment]
A quick experiment shows you can do something like `command: "./run_plugin $PWD"` and have the current directory on the command line to your plugin. 👍  for that.

(Unfortunately it splits the command line on spaces, regardless of quoting, so if your current path has a space in it you need to paste it back together again in the plugin, but it makes things possible until this is fixed.)

**Comment by jordanwilson230** [comment]
It's nasty, but one other workaround is to put a flag in your plugin and have your bashrc or bash_profile pass it the user's path.  E.g.:

function kubectl() { case $* in "deploy"*) shift 1; command kubectl plugin deploy --path=$PWD "$@" ;; * ) command kubectl "$@" ;; esac }
' >> ~/.bash_profile

Thats a snippet of what I'm currently using. Using this method is a bad approach, but will have to use until a fix is found. I put that in an install.sh script, inside of tar archive of my plugins directory.  Note that function makes it so you don't need to pass it the 'plugin' command when executing.

**Comment by jordanwilson230** [comment]
@ae-dg regarding the spacing issue, you should be able to fix that by quoting the $PWD you're passing to your script, or changing the IFS variable to handle it.

**Comment by ae-dg** [comment]
@jordanwilson230 I tried quoting $PWD but my script ended up being called with $1 being `'/tmp/a` and $2 being `dir'`. I think it's line 51 of https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/plugins/runner.go that just does a simple split().

I also tried setting command to be `CURDIR=$PWD ./myplugin` to see if I could get a new variable into the plugin. But that wouldn't execute. At the moment it looks like it's very simplistic; take the `command`, split it on spaces, the first value is the command to execute and the rest are the arguments.
